scripts/irods/test/test_quotas.py

```diff
@@ -153,3 +153,64 @@ def acRescQuotaPolicy(rule_args, callback, rei):
             self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_1'])
             self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_2'])
             IrodsController().reload_configuration()
+
+    # The following test covers case 40, 41, 42
+    def test_physical_resource_and_total_quota_simultaneously__issue_4089(self):
+        pep_map = {
+            'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                acRescQuotaPolicy {
+                    msiSetRescQuotaPolicy("on");
+                }
+            '''),
+            'irods_rule_engine_plugin-python': textwrap.dedent('''
+                def acRescQuotaPolicy(rule_args, callback, rei):
+                    callback.msiSetRescQuotaPolicy('on')
+            ''')
+        }
+
+        try:
+            with temporary_core_file() as core:
+                core.add_rule(pep_map[self.plugin_name])
+                IrodsController().reload_configuration()
+
+                group_name = 'test_group_case_40_41_42'
+                file_name = 'test_file_case_40_41_42'
+                resc_name = 'ufs0_quota_case_40_41_42'
+                lib.make_file(file_name, 4096, contents='arbitrary')
+                lib.create_ufs_resource(self.admin, resc_name)
+
+                self.admin.assert_icommand(['iadmin', 'mkgroup', f'{group_name}_1'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_1', resc_name, '5000'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_1', 'total', '10000'])
+
+                self.admin.assert_icommand(['iadmin', 'atg', f'{group_name}_1', self.user0.username])
+
+                self.user0.assert_icommand(['iput', '-R', self.testresc, file_name])
+                self.admin.assert_icommand(['iadmin', 'cu'])
+
+                # Case 40: under quota succeeds
+                with self.subTest(testcase=40, desc="Under quota before and after"):
+                    self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_40'])
+                    self.admin.assert_icommand(['iadmin', 'cu'])
+
+                # Case 41: crossing quota succeeds
+                with self.subTest(testcase=41, desc="Crossing quota"):
+                    self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_41'])
```

Comment:
I see `icp` is used, which is good/understandable. Have you thought of a plan for other icommands?

I'm not saying you have to cover them right now, just wondering if you've started developing opinions on how to tackle it and whether it's worth covering all of them.